AmplifyPlugins/Core/AWSPluginsCore/Model/GraphQLRequest/GraphQLRequest+Model.swift

```diff
@@ -24,12 +24,28 @@ extension GraphQLRequest {
     ///
     /// - seealso: `GraphQLMutation`, `GraphQLMutationType`
     public static func mutation<M: Model>(of model: M,
+                                          where predicate: QueryPredicate? = nil,
                                           type: GraphQLMutationType) -> GraphQLRequest<M> {
-        let document = GraphQLMutation(of: model, type: type)
-        return GraphQLRequest<M>(document: document.stringValue,
-                                 variables: document.variables,
-                                 responseType: M.self,
-                                 decodePath: document.decodePath)
+        switch type {
+        case .create:
+            let document = GraphQLCreateMutation(of: model, where: predicate)
+            return GraphQLRequest<M>(document: document.stringValue,
+                                     variables: document.variables,
+                                     responseType: M.self,
+                                     decodePath: document.name)
```

Comment:
If all you do is vary the `document` type in the `switch` cases, move the duplicated `return GraphQLRequest` statements outside of the switch

Reply:
oh i see why i did this, i was getting 
`Protocol 'GraphQLDocument' can only be used as a generic constraint because it has Self or associated type requirements`, not sure how to declare the 'document' type